config/confighttp/confighttp.go

```diff
@@ -114,6 +115,12 @@ type HTTPServerSettings struct {
 	// An empty list means that CORS is not enabled at all. A wildcard (*) can be
 	// used to match any origin or one or more characters of an origin.
 	CorsOrigins []string `mapstructure:"cors_allowed_origins"`
+
+	// CorsHeaders are the allowed CORS headers for HTTP/JSON requests to grpc-gateway adapter
+	// for the OTLP receiver. See github.com/rs/cors
+	// CORS needs to be enabled first by providing a non-empty list in CorsOrigins
+	// A wildcard (*) can be used to match any header.
+	CorsHeaders []string `mapstructure:"cors_allowed_headers"`
```

Comment:
In an ideal world, would you have allowed headers per origin?

Reply:
Asking because we still have an opportunity to fix this before GA, so I would like to not just add things now and have to cleanup in 2 months :)

Maybe we can get this right now and deprecate the old way.

Reply:
Good question. I think we are largely relying on [rs/cors](https://github.com/rs/cors/blob/master/cors.go) model here, which does not provide such capability. At the same time, something like this can be achieved with separate receivers (though they would need different ports, so not ideal, but when used via reverse proxy maybe not so bad either). @keitwb @kkruk-sumo - what do you think?

Reply:
@pmm-sumo based on your experience what is the more common case:
* Having same headers per multiple origins? 
* Having special headers per each origin?

Based on this we can optimize for one case or the other :)

Reply:
Unfortunately, I don't have much experience with CORS. My impression is that within a single deployment, origins can be quite similar to each other so they could share the set of headers (so option 1). But would be great to check with someone more experienced, hence the call to @keitwb and @kkruk-sumo :)

Reply:
I doubt many people will want special headers per origin for a single instance of a CORS enabled receiver.  That seems overly complicated for the config and you can just make multiple receiver instances with separate ports if you really wanted such a thing.